### ISSUE TITLE
File manager (filemin) does not honor pagination values

### ISSUE THREAD

**Issue by Mecha-Weasel** [webmin/webmin#2108]
### Environment:

- New installation of LMDE (Linux Mint Debian Edition) basically Mint built on Debian 12.x repos, etc.
- Running as VM under VMware Player 17.
- Fresh install of Webmin, installed via the setup-repos process (from here https://webmin.com/download/)
- Webmin, version: **2.105**
- Authentic theme, version: **21.09.5**

### How to recreate:

- Open webmin URL.
- Login.
- Open **File Manager** under **Tools** category.
- Select **settings** (gear) icon, in top-right corner.
- Set **Default number of items per page** option as desired, selecting from drop-down list (say **500**).
- Set **Maximum number of items in directory without using server pagination** option as desired, selecting from drop-down list (say **1000**).
- Clock **Save** button to return to **File Manager** interface.

### Expected results:

- File Manager pane should allow a long list (500 items) with vertical scroll-bar as needed.

### Actual results:

- File Manager pane is still limiting the list of items to either what fits without scrolling, or even less - forcing multiple "pages".

### Some examples:

- Viewing **/**, and showing 14 of 25 items, and making it 2 pages.
- Viewing **/etc,**, and only showing 14 of 156 items, and making it 12 pages.

### Notes:

- I check the /etc/webmin/filemin folder to see if the preferences were getting populated.
- The "per_page=500" option is being populated in the user's **perfs.userid** file.
- The "config_portable_module_filemanager_records_for_server_pagination=1000" option is being populated in the user's **perfs.userid** file.

**Comment by iliajie** [comment]
Hello,

Thanks for reaching out!

Here's what I tried based on your report (Debian 12) and the result I got:

https://github.com/webmin/webmin/assets/4426533/b059763c-65ef-4730-836c-21c7601ef323

---

Aside from everything, I would strongly advise against displaying 500 files on the page. It will make user experience horrible with everything laggy and slow. You don't need to do it. If you need to find a file, just start typing, and it will filter results you're looking for in no time.

**Comment by Mecha-Weasel** [comment]
Still not working for me.  Doesn't matter what I set the options tool, seems to do what it feels like.
Also I do not have an "automatic" option in my File Manager options page.

**Comment by Mecha-Weasel** [comment]
PS: to be more specific about OS environment, I am running LMDE (Linux Mint Debian Edition, https://linuxmint.com/download_lmde.php) which Webmin is detecting as "Debian 12".

**Comment by iliajie** [comment]
> Still not working for me. Doesn't matter what I set the options tool, seems to do what it feels like.
> Also I do not have an "automatic" option in my File Manager options page.

Which browser are you using? Did you try another browser, Chromium for example, or incognito tab with all browser extensions disabled?

**Comment by Mecha-Weasel** [comment]
yeah made no difference. but if I switch from **Authentic Theme** to **Framed Theme**, the pagination issue goes away - but of course File Manager looks completely different too.

**Comment by iliajie** [comment]
What kind of Webmin user is that? Is this safe Webmin user? What is set on File Manager ACL page (Webmin Users) exactly?

**Comment by Mecha-Weasel** [comment]
Here's what is set for that user:

![image](https://github.com/webmin/webmin/assets/3591043/3b7a4637-247e-439d-9f15-6b5602bd54de)

**Comment by iliajie** [comment]
I cannot reproduce this issue, it works just fine for me!

Though, I have a question - what is the output of the following command on your server:

```
find /etc -name *gamer-servers*
```

**Comment by Mecha-Weasel** [comment]
I'm sorry. Don't mean to be a pain, just don't understand why this happens.  I do appreciate the attempts.

here's that output:
```
root@ExampleVPS-2024b-Live:~# find /etc -name *game-servers;
/etc/webmin/shell/previous.game-servers
/etc/webmin/filemin/prefs.game-servers
/etc/webmin/custom/prefs.game-servers

```

**Comment by iliajie** [comment]
> I'm sorry. Don't mean to be a pain, just don't understand why this happens.

No problem. I'm willing to help.



> /etc/webmin/filemin/prefs.game-servers

What is the content of `/etc/webmin/filemin/prefs.game-servers` file?

**Comment by Mecha-Weasel** [comment]
BTW: This doesn't happen for that **game-servers** account, if I login to Webmin as **root**, the pagination issue is there too.

Here are the contents of that file:
```
config_portable_module_filemanager_remember_tabs=false
config_portable_module_filemanager_show_dot_files=true
config_portable_module_filemanager_hide_tree_view=false
config_portable_module_filemanager_hide_actions=true
config_portable_module_filemanager_calculate_size=true
config_portable_module_filemanager_tree_expand_search=true
config_portable_module_filemanager_hide_toolbar=false
config_portable_module_filemanager_view_limit=512000
per_page=500
config_portable_module_filemanager_hovered_toolbar=false
config_portable_module_filemanager_files_safe_mode=true
config_portable_module_filemanager_checksum_limit=1024000
config_portable_module_filemanager_tree_view_depth=3
columns=type,size,owner_user,permissions,last_mod_time
config_portable_module_filemanager_editor_maximized=false
config_portable_module_filemanager_editor_detect_encoding=true
config_portable_module_filemanager_switch_users=true
config_portable_module_filemanager_move_to_trash=false
config_portable_module_filemanager_default_sort=0
config_portable_module_filemanager_records_for_server_pagination=1000
config_portable_module_filemanager_force_tar=true
config_portable_module_filemanager_tree_exclude_on_first_load=true
config_portable_module_filemanager_editor_tabs_to_spaces=false
```

**Comment by iliajie** [comment]
Can you also run then the following and see what you get:

```
grep -Rsi per_page /etc /root
```

**Comment by Mecha-Weasel** [comment]
Here's the output ...

```
root@ExampleVPS-2024b-Live:~# grep -Rsi per_page /etc /root
/etc/webmin/filemin/prefs.game-servers:per_page=500
/etc/webmin/filemin/prefs.root:per_page=500
/etc/webmin/authentic-theme/settings-game-servers.js:config_portable_module_filemanager_auto_per_page=true;
/etc/webmin/authentic-theme/settings-game-servers.js:config_portable_module_filemanager_records_per_page='500';
/etc/webmin/authentic-theme/settings-root.js:config_portable_module_filemanager_records_per_page='20';
/etc/webmin/authentic-theme/settings-root.js:config_portable_module_filemanager_auto_per_page=true;
```

**Comment by iliajie** [comment]
> /etc/webmin/authentic-theme/settings-root.js:config_portable_module_filemanager_records_per_page='20';
> /etc/webmin/authentic-theme/settings-root.js:config_portable_module_filemanager_auto_per_page=true;

These settings are supposed to be changing on saving File Manager configuration page. Try debugging it in your browser or just manually change (i.e. `20` has to be `500` and `true` goes to `false`) those settings and reload the page.

**Comment by iliajie** [comment]
Ah, I see &mdash; you're in the Cloudflare! I wish I knew this before, as they are blocking stuff!

**Comment by Mecha-Weasel** [comment]
> config_portable_module_filemanager_auto_per_page

Cloudflare? no, I am not.  This is just a VMware Player VM running Debian, that happens to be on a LMDE desktop host.  Its all just on the LAN here at home.

**Comment by Mecha-Weasel** [comment]
Works for the **game-servers** account too (after making that change to **config_portable_module_filemanager_auto_per_page** for **game-servers** also):

![image](https://github.com/webmin/webmin/assets/3591043/0b374b8a-3329-40c5-aacf-538c04370a83)

**Comment by iliajie** [comment]
You should fix your fonts!

**Comment by Mecha-Weasel** [comment]
Is the **config_portable_module_filemanager_auto_per_page** option implemented in the theme?

**Comment by iliajie** [comment]
> Is the **config_portable_module_filemanager_auto_per_page** option implemented in the theme?

Yes, it is.

**Comment by Mecha-Weasel** [comment]
> You should fix your fonts!

Looks okay live on this end but on the preview here looks bad, but I think thats just scaling the image.

**Comment by Mecha-Weasel** [comment]
Wondering if some bug in the theme then? or some other one-time issue that messed-up my configs?  Anyway, thanks so much for the help!

**Comment by iliajie** [comment]
Are you running this instance with Cloudflare?

**Comment by Mecha-Weasel** [comment]
> Are you running this instance with Cloudflare?

No.  Literally a VMware Player VM running on a desktop computer at home.

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run.

⚠️ If the issue is resolved, please take a moment to close it to help keep the project organized.
